synapse/federation/sender/transaction_manager.py

```diff
@@ -124,6 +125,17 @@ async def send_new_transaction(
                 len(pdus),
                 len(edus),
             )
+            if issue_8631_logger.isEnabledFor(logging.DEBUG):
+                DEVICE_UPDATE_EDUS = {"m.device_list_update", "m.signing_key_update"}
+                device_list_updates = [
+                    edu.content for edu in edus if edu.edu_type in DEVICE_UPDATE_EDUS
+                ]
+                if device_list_updates:
```

Comment:
Would we want to log if the transaction doesn't include any?

Reply:
I don't think so---I'm guessing most of these EDUs are typing notifications, which I don't want to spam the logs with.

Reply:
Ah, good point! 👍